src/v0/sources/shopify/util.js

```diff
@@ -151,26 +151,32 @@ const setAnonymousIdorUserIdFromDb = async (message, metricMetadata) => {
         }
         return;
       }
-      cartToken = message.properties?.cart_token;
+      cartToken = message.properties.cart_token;
       break;
     /*
      * we dont have cart_token for carts_create and update events but have id and token field
      * which later on for orders become cart_token so we are fethcing cartToken from id
      */
     case SHOPIFY_TRACK_MAP.carts_create:
     case SHOPIFY_TRACK_MAP.carts_update:
-      cartToken = message.properties?.id;
+      cartToken = message.properties?.id || message.properties?.token;
       break;
     // https://help.shopify.com/en/manual/orders/edit-orders -> order can be edited through shopify-admin only
     // https://help.shopify.com/en/manual/orders/fulfillment/setting-up-fulfillment -> fullfillments wont include cartToken neither in manual or automatiic
     case SHOPIFY_TRACK_MAP.orders_edited:
+    case SHOPIFY_TRACK_MAP.orders_delete:
     case SHOPIFY_TRACK_MAP.fulfillments_create:
     case SHOPIFY_TRACK_MAP.fulfillments_update:
       if (!message.userId) {
         message.setProperty('userId', 'shopify-admin');
       }
       return;
     default:
+      logger.error(`Event ${message.event} not supported`);
+  }
+  if(!isDefinedAndNotNull(cartToken)){
+    message.setProperty('anonymousId', 'shopify-admin');
```

Comment:
Any reasoning for having `shopify-admin` string value for anonId as fallback ?

Reply:
The event for which cartToken is not available in the payload most probably is coming from the admin dashboard. Thats why. In addition to that if cartToken is null for any payload then it is coming from admin panel only. 
Reference:
https://community.shopify.com/c/shopify-apis-and-sdks/empty-cart-token-in-shopify-webhooks-checkout-payload/td-p/418118